### PR TITLE
assistant — fix timing issues & invalid context

### PR DESCRIPTION
**Pull request by theomonnom**
- fix invalid context with long speech 
- make sure only one speech can start at a time
- fix tts_forwarder.aclose never ending
      - also added segment_playout_finished to notify when a segment ended playing (useful for synchronizing close)
- fix _play_speech_if_validated_task never ending because playout_tx.close was never called when synthesize is cancelled
- added TaskSet to create task in a fire-and-forget fashion (you still have to aclose the TaskSet — wdyt about this the aclose fnc API?)
- fixed some misc typing stuff

### CODE REVIEW

File: livekit-agents/livekit/agents/tokenize/tokenizer.py
Comment (theomonnom) [inline livekit-agents/livekit/agents/tokenize/tokenizer.py:27]: This was never used

**Comment by ubeytd** [comment]
Great thanks 🙏🏻! When will this be released approximately?

**Comment by theomonnom** [comment]
> Great thanks 🙏🏻! When will this be released approximately?

Hey, as soon as this is approved, it may be today when the folks in the Baie Area wake up :)

**Review by keepingitneil — APPROVED** [review]

File: livekit-agents/livekit/agents/transcription/tts_forwarder.py
Comment (davidzhao) [inline livekit-agents/livekit/agents/transcription/tts_forwarder.py:247]: remove debug?

File: livekit-agents/livekit/agents/transcription/tts_forwarder.py
Comment (davidzhao) [inline livekit-agents/livekit/agents/transcription/tts_forwarder.py:243]: remove debug?

File: livekit-agents/livekit/agents/voice_assistant/assistant.py
Comment (davidzhao) [inline livekit-agents/livekit/agents/voice_assistant/assistant.py:543]: nit, spelling

```suggestion
                text=self._transcribed_text,
```

**Review by davidzhao — APPROVED** [review]
lgtm! just a couple of nits

**Pull request merged**
